pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,277 @@ import (
+	DataProtectionCapabilities StorageCohortDataProtectionCapabilities `json:"dataProtectionCapabilities,omitempty"`
+}
+
+// StorageCohortDataStorageCapabilities
+type StorageCohortDataStorageCapabilities struct {
+	// Contains the types of access modes supported by the cohort
+	AccessModes []StorageCohortAccessMode `json:"accessModes,omitempty"`
+
+	// ProvisioningPolicies defines what type of pool/volume can be created inside the cohort.
+	ProvisioningPolicies []ProvisioningPolicy `json:"provisioning_policies,omitempty"`
+
+	// Multipathing describes the different configurations of accessing
+	// storage through different nodes within a cohort
+	Multipathing StorageCohortMultipathingOption `json:"multipathing,omitempty"`
+
+	// If specified, the compression algorithm supported by the cohort
+	// +nullable
+	Compression *[]StorageCohortCompressionAlgorithm `json:"compression,omitempty"`
+
+	// Deduplication, if set to true, ensures that the data is de-duplicated within the cohort
+	// +nullable
+	Deduplication *bool `json:"deduplication,omitempty"`
+}
+
+// StorageCohortDataSecurityCapabilities
+type StorageCohortDataSecurityCapabilities struct {
+	// MediaEncryption either specifies whether encryption is supported or specifies the encryption algorithm itself
+	// +nullable
+	MediaEncryption *[]StorageCohortEncryptionAlgorithm `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy specifies the supported data sanity checks
+	DataSanitizationPolicy StorageCohortDataSanitizationPolicy `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// StorageCohortIOConnectivityCapabilities
+type StorageCohortIOConnectivityCapabilities struct {
+	// Contains the list of access protocols supported by the cohort
+	AccessProtocols []StorageCohortAccessProtocol `json:"accessProtocols,omitempty"`
+}
+
+// StorageCohortIOPerformanceCapabilities
+type StorageCohortIOPerformanceCapabilities struct {
+	// Average I/O operation latency(in microseconds) allowed within the cohort
+	AverageIOOperationLatencyMicroseconds uint64 `json:"averageIoOperationLatencyMicroseconds,omitempty"`
+
+	// Max I/O operation(per second/TB) allowed within the cohort
+	MaxIOOperationsPerSecondPerTerabyte uint64 `json:"maxIoOperationsPerSecondPerTerabyte,omitempty"`
+
+	// List of storage tiers supported by the cohort
+	StorageTier []StorageCohortStorageTier `json:"storageTier,omitempty"`
 }
 
+// StorageCohortDataProtectionCapabilities
+type StorageCohortDataProtectionCapabilities struct {
+	// Whether replication is supported by the cohort
+	// +optional
+	Replication *bool `json:"replication,omitempty"`
+
+	// Backup controls data can be backed up by the cohort
+	// +optional
+	Backup *bool `json:"backup,omitempty"`
+
+	// whether cohort will support snapshots
+	// +optional
+	Snapshots *bool `json:"snapshots,omitempty"`
+}
+
+// ComponentCondition contains condition information for a storage cohort.
+type ComponentCondition struct {
```

Comment:
Is this referred from K8s? If yes, is there a ready structure in K8s that we can use?

Reply:
Yes, its taken from k8s. But the ready structure was only specific to node and we need one more field inside this structure i.e the `Name` to specify which component condition it displays.

Reply:
Ok

Reply:
Shall we rename this to `GenericCondition` since it is being used as both node & component condition?